src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -1,21 +1,30 @@
-import { FC } from "react";
+import { FC, useEffect, useState } from "react";
 
 import { Track, Progress } from "./ProgressBar.styles";
+import getProgressBarAnimationProps from "./helpers/getProgressBarAnimationProps";
 
 const ProgressBar: FC<{
   startTime: number;
   endTime: number;
 }> = ({ startTime, endTime }) => {
-  const now = Date.now();
-  const done = now - startTime;
-  const left = endTime - startTime;
-  const fractionDone = done / left;
+  const [isMounted, setIsMounted] = useState(false);
+
+  useEffect(() => {
+    setIsMounted(true);
+  }, []);
+
+  const { progress, duration } = getProgressBarAnimationProps(
```

Comment:
naming here confused me initially - I thought `progress` would update. Wonder if this might be easier to understand:

```typescript
const { progress: initialProgress , duration } = getProgressBarAnimationProps(
```

Reply:
Yeah, initialProgress would be more fitting.